src/Components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { useState, useEffect, useRef } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { deleteTodo, updateTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo,
+  todos: Todo[];
+  setTodos: (todos: Todo[]) => void;
+  setErrorMessage: (arg: string) => void;
+  USER_ID: number;
+  setFocusedInput: (arg: boolean) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  todos,
+  setTodos,
+  setErrorMessage,
+  USER_ID,
+  setFocusedInput,
+}) => {
+  const [editing, setEditing] = useState(false);
+  const [currentIdToLoading, setCurrentIdToLoading]
+    = useState<number | null>(null);
```

Comment:
loader should be visible on each todo which now in prcess on server (when update or delete multiply todos)
for it suggest to lift up this state in App component. and it can be state with arrary of todos id which now in process.
in todoItem check whether current todo.id includes in todosInProcess state.